charts/spinnaker/templates/configmap/halyard-config.yaml

```diff
@@ -129,7 +130,29 @@ data:
                 {{ if $.Values.kubeConfig.liveManifestCalls }}--live-manifest-calls true{{ end }} \
                 --provider-version v2
     {{- end }}
-    $HAL_COMMAND config deploy edit --account-name {{ .Values.kubeConfig.deploymentContext }} --type distributed \
+    {{- end }}
+
+    {{- if $.Values.kubeConfig.accountPerContextAndNamespace }}
+    {{- range $context := .Values.kubeConfig.contexts }}
```

Comment:
Why isn't `.Values.kubeConfig.contexts` prefixed with `$` like in other places. I know by using `$` we can use the root scope but I'm interested in knowing the difference it makes here 😅

Reply:
Good catch. This indeed has to do with the scope we are in. Here, it is not necessary to specify `$` as we are still in the 'root'. If we do `range`, we enter a new scope. One line below, we are in scope of `context` and therefore we need to specify `$.Values....` to refer to the root. To that sense, the usage of `$` in the `if` statements can be omitted, I think.

Reply:
Thanks for the explanation and now it is making sense.

> To that sense, the usage of $ in the if statements can be omitted, I think.

I agree. It would be nice to be consistent 🙂

Reply:
Done!